rstudio/rstudio
# Add the ability to create ED25519 SSH keys, and set ED25519 as the default

### Intent

Addresses #8255.

- Adds support for ED25519 encrypted keys to be created (by default)
- Increase size for RSA keys so if they are created, they follow Github recommendation (4096 bytes)
- Update all dialogs/labels that refer to RSA SSH keys to refer more generally to SSH keys

### Approach

RPC call already had an argument for SSH key type to be passed along, it was just set to "rsa" as a constant
Updated so user can choose their preferred key type when they create a new key. As before, when searching for existing keys, we check for ED25519 keys first, and then look for RSA only if not found -- change is only to enable the creation of new ED25519 keys (and better RSA keys) from within the IDE

### Automated Tests

None

### QA Notes

- Create a new ED25519 key
- Create a new RSA key
- Check which key is detected if both key types exist
- Check which key is detected if only one key type exists (eg should detect RSA if user has only RSA)

(You may have to restart RStudio to re-detect deleted keys)

### Checklist
